vyper/semantics/validation/annotation.py

```diff
@@ -260,6 +260,15 @@ def visit_Subscript(self, node, type_):
 
     def visit_Tuple(self, node, type_):
         node._metadata["type"] = type_
+
+        if type_ is not None and not isinstance(type_, BaseTypeDefinition):
+            # some tuples are straight type annotations e.g. `(String[100], uint256)`
+            # in `_abi_decode(x, (String[100], uint256))`.
+            # skip annotating them because they do not conform to
+            # the BaseTypeDefinition API (and anyways we do not need to
+            # annotate them!)
+            return
+
```

Comment:
i think first of all maybe the check should be `if isinstance(type_, TypeTypeDefinition)`

and then make the check in `visit_Name` into the same form so that somebody reading the code sees consistency of the usage